tests/integration/route_test.py

```diff
@@ -0,0 +1,208 @@
+                        'enabled': True
+                    }
+                }
+            ],
+            'routes': {
+                'config': [
+                    IPV4_ROUTE1,
+                    IPV4_ROUTE2,
+                    IPV6_ROUTE1,
+                    IPV6_ROUTE2,
+                ]
+            },
+        })
+    cur_state = statelib.show_only(('eth1',))
+    _assert_has_route(IPV4_ROUTE1, cur_state)
+    _assert_has_route(IPV4_ROUTE2, cur_state)
+    _assert_has_route(IPV6_ROUTE1, cur_state)
+    _assert_has_route(IPV6_ROUTE2, cur_state)
+
+
+@pytest.mark.xfail(raises=NmstateNotImplementedError,
+                   reason="Red Hat Bug 1707396")
+def test_multiple_gateway(eth1_up):
+    netapplier.apply(
+        {
+            'interfaces': [
+                {
+                    'name': 'eth1',
+                    'state': 'up',
+                    'type': 'ethernet',
+                    'ipv4': {
+                        'address': [
+                            {
+                                'ip': IPV4_ADDRESS1,
+                                'prefix-length': 24
+                            }
+                        ],
+                        'dhcp': False,
+                        'enabled': True
+                    },
+                    'ipv6': {
+                        'address': [
+                            {
+                                'ip': IPV6_ADDRESS1,
+                                'prefix-length': 64
+                            }
+                        ],
+                        'dhcp': False,
+                        'autoconf': False,
+                        'enabled': True
+                    }
+                }
+            ],
+            'routes': {
+                'config': [
+                    IPV4_GATEWAY1,
+                    IPV4_GATEWAY2,
+                    IPV6_GATEWAY1,
+                    IPV6_GATEWAY2,
+                ]
+            },
+        })
+    cur_state = statelib.show_only(('eth1',))
+    _assert_has_route(IPV4_GATEWAY1, cur_state)
+    _assert_has_route(IPV4_GATEWAY2, cur_state)
+    _assert_has_route(IPV6_GATEWAY1, cur_state)
+    _assert_has_route(IPV6_GATEWAY2, cur_state)
+
+
+def _assert_has_route(route, state):
+    assert route in state['routes']['config']
+
+    # For running route, the default(-1) metric and route-table will be
+    # changed. Hence should be ignored.
+    tmp_route = copy.deepcopy(route)
+    run_routes = copy.deepcopy(state['routes']['running'])
+    for prop_name, default_value in (('metric', -1), ('table-id', 0)):
+        if tmp_route[prop_name] == default_value:
+            del tmp_route[prop_name]
+            for run_route in run_routes:
+                del run_route[prop_name]
+    assert tmp_route in run_routes
+
+
+def _assert_not_has_route(route, state):
+    for routes in (state['routes']['running'], state['routes']['config']):
+        assert route not in routes
```

Comment:
Why does this not need to handle metric and table-id in a special way as the `_assert_has_route()` function does?

Reply:
Yeah. I should do the same. But I have removed `_assert_has_route()` from this PR as it's not used.